agent/engine/dependencygraph/graph.go

```diff
@@ -287,10 +276,22 @@ func containerOrderingDependenciesCanResolve(target *apicontainer.Container,
 			// The 'target' container desires to be moved to its 'steady' state.
 			// Allow this only if the dependency container also desires to be in 'steady state'
 			return dependsOnContainerDesiredStatus == dependsOnContainer.GetSteadyStateStatus()
-
 		}
 		return false
 
+	case "SUCCESS":
+		isDependOnContainerStoppedSuccessfully := dependsOnContainer.GetKnownStatus() == apicontainerstatus.ContainerStopped &&
+			*dependsOnContainer.GetKnownExitCode() == 0
+		return verifyContainerOrderingStatus(target, dependsOnContainer) || isDependOnContainerStoppedSuccessfully
+
+	case "COMPLETE":
+		isDependOnContainerStopped := dependsOnContainer.GetKnownStatus() == apicontainerstatus.ContainerStopped &&
+			dependsOnContainer.GetKnownExitCode() != nil
```

Comment:
do we need this, or is it implied by ContainerStopped?

Reply:
I think probably we can get rid of this. Initially I added this, to cover all corner cases but I thi nk we can get rid of this